### PR TITLE
#611: Fixed a crash using opaque data types

### PR DESCRIPTION
**Pull request by davidcok**

### CODE REVIEW

File: Source/Dafny/DafnyAst.cs [hunk 1]
```diff
@@ -755,9 +755,12 @@ public abstract class Type {
           if (!rtd.AsTopLevelDecl.IsVisibleInScope(scope)) {
             // This can only mean "rtd" is a class/trait that is only provided, not revealed. For a provided class/trait,
             // it is the non-null type declaration that is visible, not the class/trait declaration itself.
+            // rtd can also be an OpaqueTypeDecl
```
Comment (RustanLeino) [inline Source/Dafny/DafnyAst.cs:758]: To confirm this belief, I suggest adding an `else` branch to the `if` below, asserting that `rtd` is an `OpaqueTypeDecl`.

File: package.py [hunk 2]
```diff
@@ -47,23 +47,24 @@
...
+        "Dafny",
+        "DafnyRuntime",
+        "DafnyPipeline",
         "Mono.Cecil",
+        "Provers.SMTLib",
         "System.Collections.Immutable",
         "System.Runtime"]
 EXEs = ["Dafny", "DafnyServer"]
-ETCs = UNIX_EXECUTABLES + ["DafnyPrelude.bpl", "DafnyRuntime.cs", "DafnyRuntime.js", "DafnyRuntime.go", "DafnyRuntime.jar", "libhostpolicy.dylib"]
+ETCs = UNIX_EXECUTABLES + ["DafnyPrelude.bpl", "DafnyRuntime.cs", "DafnyRuntime.js", "DafnyRuntime.go", "DafnyRuntime.jar", "Dafny.deps.json", "Dafny.runtimeconfig.json"]
```
Comment (RustanLeino) [inline package.py:67]: Is `package.py` meant to be part of this PR?

**Review by RustanLeino — CHANGES_REQUESTED** [review]

File: Source/Dafny/DafnyAst.cs (hunk 1 above)
Comment (davidcok) [reply Source/Dafny/DafnyAst.cs]: Done

File: package.py (hunk 2 above)
Comment (davidcok) [reply package.py]: No - fixed it

**Pull request merged**